src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs

```diff
@@ -1335,6 +1335,96 @@ public async Task ServerThrowsHubExceptionIfBuildingAsyncEnumeratorIsNotPossible
             }
         }
 
+        [Theory]
+        [QuarantinedTest]
+        [MemberData(nameof(HubProtocolsList))]
+        public async Task ServerLogsErrorIfClientInvokeCannotBeSerialized(string protocolName)
+        {
+            // Just to help sanity check that the right exception is thrown
+            var exceptionSubstring = protocolName switch
+            {
+                "json" => "A possible object cycle was detected.",
+                "newtonsoft-json" => "A possible object cycle was detected.",
+                "messagepack" => "Failed to serialize Microsoft.AspNetCore.SignalR.Client.FunctionalTests.TestHub+Unserializable value.",
+                var x => throw new Exception($"The test does not have an exception string for the protocol '{x}'!"),
+            };
+
+            var protocol = HubProtocols[protocolName];
+            using (var server = await StartServer<Startup>(write => write.EventId.Name == "FailedWritingMessage"))
+            {
+                var connection = CreateHubConnection(server.Url, "/default", HttpTransportType.LongPolling, protocol, LoggerFactory);
+                var closedTcs = new TaskCompletionSource<Exception>();
+                connection.Closed += (ex) => { closedTcs.TrySetResult(ex); return Task.CompletedTask; };
+                try
+                {
+                    await connection.StartAsync().OrTimeout();
+
+                    var result = connection.InvokeAsync<string>(nameof(TestHub.CallWithUnserializableObject)).OrTimeout();
+
+                    // The connection should close.
+                    await closedTcs.Task.OrTimeout();
+                }
+                catch (Exception ex)
+                {
+                    LoggerFactory.CreateLogger<HubConnectionTests>().LogError(ex, "{ExceptionType} from test", ex.GetType().FullName);
```

Comment:
Do we expect this catch block to run? It looks to me like if `await closedTcs.Task.OrTimeout();` times out this would swallow it and the test would still pass. Is that not the case?

Reply:
My bad, there's a rethrow in the other tests that do this. I removed it to do some testing and didn't put it back ;)